api/lib/messageCreators/polkadot-transactions.js

```diff
@@ -160,30 +160,28 @@ async function RestakeTx(senderAddress, api, { to, from, addressRole }) {
   return await getSignMessage(senderAddress, transactions, api)
 }
 
-async function ClaimRewardsTx(senderAddress, api) {
+async function ClaimRewardsTx(
+  senderAddress,
+  api,
+  { rewards }
+) {
   let allClaimingTxs = []
-  const stakerRewards = await api.derive.staking.stakerRewards(senderAddress)
-  const newStakerRewards = stakerRewards.filter(({ era }) => era.toJSON() > 718)
-  if (newStakerRewards.length === 0) {
+
+  if (rewards.length === 0) {
     allClaimingTxs = []
   } else {
-    newStakerRewards.forEach((reward) => {
-      reward.nominating.forEach((nomination) => {
-        if (reward.isStakerPayout) {
-          allClaimingTxs.push(
-            api.tx.staking.payoutStakers(nomination.validatorId, reward.era)
+    rewards
+      .sortBy((a, b) => a.height - b.height)
+      .forEach((reward) => {
+        allClaimingTxs.push(
+          api.tx.staking.payoutStakers(
+            reward.validator.operatorAddress,
+            reward.height
           )
-        } else {
-          const validators = reward.nominating.map(
-            ({ validatorId, validatorIndex }) => [validatorId, validatorIndex]
-          )
-          allClaimingTxs.push(
-            api.tx.staking.payoutNominator(reward.era, validators)
-          )
-        }
+        )
       })
-    })
   }
+
   if (allClaimingTxs.length === 0) {
     throw new Error('There are no claimable rewards')
   }
```

Comment:
When is this function from API used actually? I just tried it and was using the `polkadot.js` file in the signing folder in FE

Reply:
ah, for network fees, I see it. It worked quite well actually

Reply:
I opened an issue to remove this file and use the APP file to reduce code repetition #4879